sql/mysqld.cc

```diff
@@ -1729,28 +1729,14 @@ static void close_connections(void)
 
   /* Abort listening to new connections */
   DBUG_PRINT("quit",("Closing sockets"));
-  if (!opt_disable_networking )
+  for (std::vector<MYSQL_SOCKET>::iterator sock = listen_sockets.begin();
+       sock != listen_sockets.end(); ++sock)
   {
-    if (mysql_socket_getfd(base_ip_sock) != INVALID_SOCKET)
-    {
-      (void) mysql_socket_close(base_ip_sock);
-      base_ip_sock= MYSQL_INVALID_SOCKET;
-    }
-    if (mysql_socket_getfd(extra_ip_sock) != INVALID_SOCKET)
-    {
-      (void) mysql_socket_close(extra_ip_sock);
-      extra_ip_sock= MYSQL_INVALID_SOCKET;
-    }
+    (void) mysql_socket_close(*sock);
+    if (sock->is_unix_domain_socket)
+      (void) unlink(mysqld_unix_port);
   }
 
```

Comment:
when do you need to `listen_sockets.clear()` after closing and when not?

Reply:
Added `listen_sockets.clear()` here, and removed `listen_sockets.clear()` from `mysql_init_variables()`.

This particular place is called only once, on shutdown. So it doesn't matter is `listen_sockets` is cleared here or not. However, in `close_server_sock()` it must be cleared, as `close_server_sock()` can be called more than once. This may be confusing, so I added `listen_sockets.clear()` here too, even if it's not needed.
Another possibly confusing place is clearing `listen_sockets` in `mysql_init_variables()`. At that time `listen_sockets` is already empty, so that call was removed.